### PR TITLE
[Windows] Hide/show titlebar on modal pages

### LINKED ISSUE #25964: [WinUI] App TitleBar overlapping on .NET9 when using Modals
### Description

1) Start Controls.Sample App
2) Core => Modal
3) Push Modal Page

Image

### PR DESCRIPTION
**Pull request by Foda**
### Description of Change

Toggle the titlebar visibility when modal pages are pushed/popped.

### Issues Fixed

Fixes #25964

### CODE REVIEW

**Review by PureWeen — DISMISSED** [review]

**Comment by mattleibow** [comment]
Can we get tests for this one?

**Comment by PureWeen** [comment]
> Can we get tests for this one?

we could probably validate the raw control.
The screen shot tests don't work because we cut off the titlebar so we can use screen comparison tests :-/ 

We're going to add a manual test as well here for validation

**Comment by jsuarezruiz** [comment]
> Can we get tests for this one?



> > Can we get tests for this one?
> 
> we could probably validate the raw control. The screen shot tests don't work because we cut off the titlebar so we can use screen comparison tests :-/
> 
> We're going to add a manual test as well here for validation

We could maybe manage the cutting logic, to avoid it when necessary. But, yeah, with a Device Test should be enough.

File: src/Controls/src/Core/Platform/ModalNavigationManager/ModalNavigationManager.Windows.cs [hunk 1]
```diff
@@ -104,13 +104,18 @@ void SetCurrent(
 					newPage.Toolbar ??= new Toolbar(newPage);
 					_ = newPage.Toolbar.ToPlatform(modalContext);
 
-					var windowManager = modalContext.GetNavigationRootManager();
-
-					if (windowManager.RootView is WindowRootView wrv)
+					// Hide titlebar on previous page
+					var previousContext = previousPage.FindMauiContext();
```
Comment (jsuarezruiz) [inline src/Controls/src/Core/Platform/ModalNavigationManager/ModalNavigationManager.Windows.cs:108]: Could include a device test? https://github.com/dotnet/maui/tree/main/src/Controls/tests/DeviceTests

Comment (Foda) [reply src/Controls/src/Core/Platform/ModalNavigationManager/ModalNavigationManager.Windows.cs:108]: Sure -- lemme add that to the existing modal push/pop Windows tests

**Review by PureWeen — APPROVED** [review]

**Review by jfversluis — APPROVED** [review]

**Pull request merged**